homepage/homepage/content/docs/using-covalues/imagedef/react.mdx

```diff
@@ -78,194 +82,229 @@ The `createImage()` function:
+
+<Image imageId="123" width="600" height="original" />
+// <img width="600" height="338" /> keeping the aspect ratio
+
+<Image imageId="123" width="original" height="600" />
+// <img width="1067" height="600" /> keeping the aspect ratio
+
+<Image imageId="123" width="600" height="600" />
+// <img width="600" height="600" />
 ```
 </CodeGroup>
 
-## Understanding ImageDefinition
+If the image was generated with progressive loading, the `width` and `height` props will determine the best resolution to use.
 
-Behind the scenes, `ImageDefinition` is a specialized CoValue that stores:
+### Imperative usage
 
-- The original image dimensions (`originalSize`)
-- An optional placeholder (`placeholderDataURL`) for immediate display
-- Multiple resolution variants of the same image as [`FileStream`s](../using-covalues/filestreams)
-
-Each resolution is stored with a key in the format `"widthxheight"` (e.g., `"1920x1080"`, `"800x450"`).
+Like other CoValues, `ImageDefinition` can be used to load the object.
 
 <CodeGroup>
-```ts twoslash
+```tsx
 import { ImageDefinition } from "jazz-tools";
-// ---cut---
-// Structure of an ImageDefinition
-const image = ImageDefinition.create({
-  originalSize: [1920, 1080],
-  placeholderDataURL: "data:image/jpeg;base64,/9j/4AAQSkZJRg...",
+
+const image = await ImageDefinition.load("123", {
+  resolve: {
+    original: true,
+  },
 });
 
-// Accessing the highest available resolution
-const highestRes = ImageDefinition.highestResAvailable(image);
-if (highestRes) {
-  console.log(`Found resolution: ${highestRes.res}`);
-  console.log(`Stream: ${highestRes.stream}`);
-}
+console.log({
+  originalSize: image.originalSize,
+  placeholderDataUrl: image.placeholderDataUrl,
+  original: image.original, // this FileStream may be not loaded yet
+});
 ```
 </CodeGroup>
 
-For more details on using `ImageDefinition` directly, see the [VanillaJS docs](/docs/vanilla/using-covalues/imagedef).
+`image.original` is a `FileStream` and its content can be read as described in the [FileStream](/docs/react/using-covalues/filestreams#reading-from-filestreams) documentation.
 
-### Fallback Behavior
+Since FileStream objects are also CoValues, they must be loaded before use. To simplify loading, if you want to load the binary data saved as Original, you can use the `loadImage` function.
 
-`highestResAvailable` returns the largest resolution that fits your constraints. If a resolution has incomplete data, it falls back to the next available lower resolution.
+<CodeGroup>
+```ts
+import { loadImage } from "jazz-tools/media";
+
+const image = await loadImage(imageDefinitionOrId);
+
+console.log({
+  width: image.width,
+  height: image.height,
+  image: image.image,
+});
+```
+</CodeGroup>
+
+If the image was generated with progressive loading, and you want to access the best resolution, `loadImageBySize` function is here the utility function you need. It will load the image of the best resolution that fits the wanted width and height.
```

Comment:
```suggestion
If the image was generated with progressive loading, and you want to access the best-fit resolution, use `loadImageBySize`. It will load the image of the best resolution that fits the wanted width and height.
```